log-levels.md

```diff
@@ -0,0 +1,54 @@
+# Log Levels 
+
+This guide serves as guideline for library authors with regards to what [SwiftLog](https://github.com/apple/swift-log) log levels are apropriate for use in libraries, and in what siuations to use what level.
+
+## Guidelines for Libraries
+
+SwiftLog defines the following 7 log levels: 
+
+* `trace`
+* `debug`
+* `info`
+* `notice`
+* `warning`
+* `error`
+* `critical`
+
+Out of those, only levels _lower than_ info (exclusive) are generally okay to be used by libraries.
+In the following section we'll explore how to use them in practice.
+
+### Recommended log levels
+
+It is always fine for a library to log at `trace` and `debug` levels, and these two should be the primary levels any library is logging at.
+
+`trace` is the finest log level, and end-users of a library will not usually use it unless debugging very specific issues. You should consider it as a way for library developers to "log everything we could possibly need to diagnose a hard to reproduce bug".
+
+**// TODO: should we suggest that a system should be able to survive with debug logging turned on in production (ktoso: I'd suggest that, yes).**
+
+This is in contrast with `debug` which some users _may_ choose to run with enabled on their production systems. As such, debug logging should not be overly noisy. It should provide a high value understanding of what is going on in the library for end users without diving deep into its internals. This is what `trace` is intended for.
```

Comment:
could be helpful to give a concrete example or two here to illustrate the distinction b/w debug and trace

Reply:
Good feedback, thanks will do!